src/components/text-input.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { css } from '@emotion/react';
+import { mediaQuery, pxToVw } from 'styles/media';
+import color from 'styles/colors';
+
+const inputText = css`
+  width: ${pxToVw(296)};
+  height: ${pxToVw(48)};
+  border: none;
+  border-bottom: 1px solid ${color.border.gray01};
+  padding: ${pxToVw(14)} 0;
+  box-sizing: border-box;
+  font-size: ${pxToVw(28)};
+
+  &:focus {
+    outline: none;
+    border-bottom: 2px solid ${color.purple};
+  }
+
+  ${mediaQuery(640)} {
+    width: 560px;
+    height: 48px;
+    padding: 14px 0;
+    font-size: 28px;
+  }
+`;
+
+const subtext = css`
+  font-size: ${pxToVw(12)};
+  color: ${color.font.gray05};
+  margin-top: 1.6vw;
+  align-self: flex-start;
+
+  ${mediaQuery(640)} {
+    font-size: 12px;
+    margin-bottom: 6px;
+  }
+`;
+
+interface InputBoxProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  label?: string;
+  note?: string;
+}
+
+export default function textInput({ label, note, ...rest }: InputBoxProps) {
+  return (
+    <>
+      <input css={inputText} type="text" {...rest} />
+      {note && <p css={subtext}>{note}</p>}
```

Comment:
`note` 값이 `false` 가 들어오면 `false` 가 반환 될텐데 사용하는 쪽에서 타입이 `false | Element` 가 돼서 타입이 점점 확장 되고, 대표적으로 `PropsWithChildren`으로 선언된 `children`으로 사용 될 경우엔 의 경우엔 타입 에러를 반환합니다. 삼항연산자로 `null` or `<></>` 을 반환 하시면 안정된 타입을 가져 갈 수 있을거 같아요!
```suggestion
      {note ? <p css={subtext}>{note}</p> : <></>}
```

Reply:
`note` 가 nullish한 상황을 의도하신 것이라면 `note == null`로 좋은 방법입니다. 항등연산자로는 `null`과 `undefined`를 동시에 발라낼 수 있어서용 ㅎㅎ

근데 `note`의 타입이 `string?`이라서 `null`이 들어올 수는 없으니 `note === null`도 괜찮을 것 같습니다!

Reply:
그러면 위 방법들과 
```jsx
{note ?? <p css={subtext}>{note}</p>}
```
이 방법 간에도 차이가 있을까요?!

Reply:
```tsx
{note ?? <p css={subtext}>{note}</p>}
```
는
```tsx
{note === null ? <p css={subtext}>{note}</p> : null}
```
또는
```tsx
{note === undefined ? <p css={subtext}>{note}</p> : undefined}
```
와 완전 동일합니당 ㅎㅎ
근데 저번에 회사에서 nullish coalescing으로 처리하려고 했다가 무슨 에러 만났던 것 같은데...잘 기억이...
잘 돌아가면 nullish coalescing 써주는 것이 훨씬 깔끔할 것 같아요!